reactor-netty-http/src/main/java/reactor/netty/http/server/logging/AccessLogArgProviderH1.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2011-Present VMware, Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.netty.http.server.logging;
+
+import io.netty.handler.codec.http.HttpRequest;
+
+import java.net.SocketAddress;
+import java.util.Objects;
+
+/**
+ * @author limaoning
+ */
+final class AccessLogArgProviderH1 extends AbstractAccessLogArgProvider<AccessLogArgProviderH1> {
+
+	HttpRequest request;
+
+	AccessLogArgProviderH1(SocketAddress remoteAddress) {
+		super(remoteAddress);
+	}
+
+	AccessLogArgProviderH1 request(HttpRequest request) {
+		this.request = Objects.requireNonNull(request, "request");
+		onRequest();
+		return self();
+	}
+
+	@Override
+	public CharSequence method() {
+		return request == null ? null : request.method().name();
+	}
+
+	@Override
+	public CharSequence uri() {
+		return request == null ? null : request.uri();
+	}
+
+	@Override
+	public String protocol() {
+		return request == null ? null : request.protocolVersion().text();
+	}
+
```

Comment:
Can we move those in the abstract class and then in the children to override the `onRequest` method where we will initialise `method`, `uri` and `protocol`